### PR TITLE
Adds alerting email address to RHMI installation CRD

### PR DESCRIPTION
**Pull request by jjaferson**
# Description

Adds `AlertingEmailAddress` field to RHMI installation CRD and uses the value present in `ALERTING_EMAIL_ADDRESS` env variable as default when the operator creates its own CRD.

https://issues.redhat.com/browse/INTLY-7406

## Verification steps
1) Run the operator against a cluster using `make cluster/prepare/local && make code/run` 
2) Verify if the RHMI CRD has the `AlertingEmailAddress` field 
3) Check if the email address was added in the `alertmanager-application-monitoring` secret in the `redhat-rhmi-middleware-monitoring-operator` namespace

```
receivers:
  - name: default
    email_configs:
      - send_resolved: true
        to: noreply-test@rhmi.com.br
```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

**Comment by KevFan** [comment]
Verified
/lgtm

File: pkg/controller/installation/installation_controller_test.go [hunk 1]
```diff
@@ -99,6 +99,64 @@ func testCreateInstallationCR_useClusterStorage(t *testing.T, envValue string, a
...
+	defer os.Unsetenv("WATCH_NAMESPACE")
+
+	// Function to test
+	err := createInstallationCR(ctx, mockClient)
+
+	if err != nil {
+		t.Fatalf("Error creating installation CR: %v", err)
+	}
+
+	installation, err := getInstallationCR(ctx, mockClient, t)
```
Comment (dimitraz) [inline pkg/controller/installation/installation_controller_test.go:122]: Should `err` be handled? The test will panic if a `nil` cr is returned for whatever reason from this function

File: pkg/controller/installation/installation_controller_test.go [hunk 2]
```diff
@@ -99,6 +99,64 @@ func testCreateInstallationCR_useClusterStorage(t *testing.T, envValue string, a
...
```
Comment (dimitraz) [inline pkg/controller/installation/installation_controller_test.go:150]: Same as above

File: pkg/controller/installation/installation_controller_test.go (hunk 1 above)
Comment (jjaferson) [reply pkg/controller/installation/installation_controller_test.go]: Good catch, thanks

**Comment by jjaferson** [comment]
/test e2e

timed out during installation waiting for completion of cloud-resources

**Pull request merged**